# HG changeset patch
# User ARostovtsev
# Parent  96d8479d68ad2a8022ca60da7e7613bc5c67e82d
#DEVBO-19 Некорректная проверка обязательности тегов level-abs-me-wtr level-abs-me. TH 16.0 #BTS-249 #BTS-250

https://msk-exp-jira.expertek.local/browse/DEVBO-19
https://msk-exp-jira.expertek.local/browse/BTS-249
https://msk-exp-jira.expertek.local/browse/BTS-250

diff --git a/ibs/th/bge/1crn/subjects/measuring.cls b/ibs/th/bge/1crn/subjects/measuring.cls
--- a/ibs/th/bge/1crn/subjects/measuring.cls
+++ b/ibs/th/bge/1crn/subjects/measuring.cls
@@ -166,17 +166,18 @@ class ibs.th.bge.1crn.subjects.measuring
 
   method override protected void CheckData ():
     def var v-str as char no-undo.
-    
     assign
       v-str = ",code"                      when code_             = ?
       v-str = v-str + ",deg-auto"          when deg-auto          = ?
       v-str = v-str + ",name"              when name_             = ?
       v-str = v-str + ",level-ms-type"     when level-ms-type     = ? and dimension-level   = 1
       /* v-str = v-str + ",level-tk"          when level-tk          = ? and dimension-level = 1  Отключен контроль обязательности в сообщении 14.05.2021 */ 
-      v-str = v-str + ",level-abs-me"      when level-abs-me      = ? and level-rel-me-np   = ? and dimension-level = 1
-      v-str = v-str + ",level-rel-me-np"   when level-abs-me      = ? and level-rel-me-np   = ? and dimension-level = 1
-      v-str = v-str + ",level-abs-me-wtr"  when level-abs-me-wtr  = ? and level-rel-me-wtr  = ? and dimension-level = 1
-      v-str = v-str + ",level-rel-me-wtr"  when level-abs-me-wtr  = ? and level-rel-me-wtr  = ? and dimension-level = 1
+      v-str = v-str + ",level-abs-me"      when level-abs-me      = ? and dimension-level = 1 
+              and (deg-auto = enum-deg-auto:GetEnum(0) or deg-auto = enum-deg-auto:GetEnum(2))
+/*      v-str = v-str + ",level-rel-me-np"   when level-abs-me      = ? and level-rel-me-np   = ? and dimension-level = 1*/
+      v-str = v-str + ",level-abs-me-wtr"  when level-abs-me-wtr  = ? and dimension-level = 1 
+              and (deg-auto = enum-deg-auto:GetEnum(0) or deg-auto = enum-deg-auto:GetEnum(2))
+/*      v-str = v-str + ",level-rel-me-wtr"  when level-abs-me-wtr  = ? and level-rel-me-wtr  = ? and dimension-level = 1*/
       v-str = v-str + ",temp-abs-me-vlm"   when temp-abs-me-vlm   = ? and dimension-temp    = 1
       v-str = v-str + ",temp-abs-me-dnst"  when temp-abs-me-dnst  = ? and dimension-temp    = 1 
       v-str = v-str + ",dnst-ms-type"      when dnst-ms-type      = ? and dimension-dnst    = 1
